app/controllers/discourse_subscriptions/admin/subscriptions_controller.rb

```diff
@@ -49,6 +51,18 @@ def destroy
           render_json_error e.message
         end
       end
+
+      private
+
+      # this will only refund the most recent subscription payment
+      def refund_subscription(subscription_id)
+        subscription = ::Stripe::Subscription.retrieve(subscription_id)
+        invoice = ::Stripe::Invoice.retrieve(subscription[:latest_invoice]) if subscription[:latest_invoice]
+        payment_intent = invoice[:payment_intent] if invoice[:payment_intent]
+        refund = ::Stripe::Refund.create({
+          payment_intent: payment_intent,
```

Comment:
What happens if `payment_intent` is nil?

Reply:
The call will fail which will recover an error to the client from the rescue on line 50. Though the likelihood of it getting into this state is very low as if an invoice exists, a payment intent exists.

Reply:
Oops, I missed the rescue.